Development

MarshalCypher opened the pull request:
Finished editing the fifth page (for speed input and display handling) and the main page to match the changes on the fifth page accordingly.
Also, a little part of the adjustments on the main page is for the sixth page that is in progress

MarshalCypher committed:
Main page Update

MarshalCypher committed:
Fifth page Update

Modified to add speed page and handle display based on test types

MarshalCypher committed:
Arduino nano update

Added a second load cell for torsion force measurement and AS5600 sensor for the angle of twist measurement then handled the serial communication between the nano and the esp32 sixth page code

MarshalCypher committed:
Nano code update

Just declared the data type for initial angle

HayBeeCoder commented on src/main.cpp:
```diff
@@ -278,8 +284,20 @@ void loop()
+      
+      input_value = TARGET_SPEED;
+    }
+    fifth_page_ui(u8g2, SELECTED_PAGE, SELECTED_PAGE_ADDRESS, key, selectedTest(), SPEED_ADDRESS, input_value, SENSOR_RATING_IN_KG, TARGET_FORCE, SELECTED_PAGE, TARGET_EXTENSION, -1, TARGET_SPEED, "mm/min");
```
instead of 'mm/min', it ought to still be EXTENSOMETER_RATING_IN_MM

HayBeeCoder replied:
The last parameter to fifth_page_ui function is expected to be the string value of the max value that the device measuring target speed can measure and not the unit of measurement.

i mean instead of "mm/min" it is meant to be its particular value say "20".

 When you test the code in the machine, on the screen, the target speed will have a value beside the exact target value. 
when you observe that , then it is the above issue i mentioned that happened. You can changing now or wait till then to see and you will correct accordingly.

HayBeeCoder commented on lib/pages/fifth_page/fifth_page.cpp:
```diff
@@ -59,16 +70,51 @@ void fifth_page_ui(U8G2_ST7920_128X64_F_SW_SPI u8g2, int page, int page_value_ad
         u8g2.drawBox(0, 10, 128, 12);
         u8g2.setDrawColor(2);
 
+        // Modify display based on test type
```
Add the following above line 73 in fifth_page.cpp:
u8g2.setFont(u8g2_font_4x6_tf);

HayBeeCoder replied:
Try and do this: 

Add the following above line 73 in fifth_page.cpp:
u8g2.setFont(u8g2_font_4x6_tf);

HayBeeCoder commented on other_microcontrollers_code/working_nano_code/working_nano_code.ino:
```diff
@@ -46,71 +65,174 @@ unsigned long timeout = 1000;  // Timeout in microseconds
-  Serial.println("weightInNewtons : ");  // readCaliper()
-  Serial.println(weightInNewtons);       // readCaliper()
-  // Serial.println("\n");
-  mySerial.print((float)myInt);   // Send weight
```
if(currenttest == torsion){
// where the myInt below is te one for 
 //torsion.
  mySerial.print((float)myInt);   
  mySerial.print(",");            
  mySerial.print(angle); 
  mySerial.println(); 
}else{
  mySerial.print((float)myInt);   
  mySerial.print(",");            
  mySerial.print(readCaliper()); 
  mySerial.println(); 
}

MarshalCypher committed:
CHANGES FROM COMMENT ALREADY MADE

MarshalCypher committed:
UPDATE on control

MarshalCypher commented:
I have edited the atmega code to control motorwith the L298n motor driver when the torsion test is selected and almost done with the sixth page that handles what value the esp would send

MarshalCypher committed:
Update on sixth page

Finished working on the sixth page handler for torsion and the other tests status

MarshalCypher committed:
Fully updated sixth page handler

Edited the status codes to be sent to the atmega

HayBeeCoder reviewed (changes requested):
try and check my comments

MarshalCypher committed:
Changes acording to comments made

Last parameter of the fifth_page_ui set to the max value and u8g2.setFont(u8g2_font_4x6_tf); line  added to the fifth page.cpp

MarshalCypher commented:
Last parameter of the fifth_page_ui set to the max value which is 150 and u8g2.setFont(u8g2_font_4x6_tf); line  added to the fifth page.cpp
